src/parser/shared/modules/items/GemChecker.tsx

```diff
@@ -0,0 +1,288 @@
+
+    //#region Special Cases
+    if (item.icon === 'inv_siren_isle_ring.jpg') {
+      return (
+        <Trans id="shared.GemChecker.CyrceSpecialCase">
+          Cyrce's Circlet is a special case. Please see your class guides for best usage.
+        </Trans>
+      );
+    }
+
+    //Not Gemable
+    if (
+      item.bonusIDs === undefined &&
+      !ItemHelper.hasBonusId(item, ItemHelper.SINGLE_GEM_BONUS_ID) &&
+      !ItemHelper.hasBonusId(item, ItemHelper.DOUBLE_GEM_BONUS_ID) &&
+      !ItemHelper.hasBonusId(item, ItemHelper.TRIPLE_GEM_BONUS_ID) &&
+      !GemChecker.twoAddableGemSlots.includes(slotNumber) &&
+      !GemChecker.oneAddableGemSlot.includes(slotNumber)
+    ) {
+      return <Trans id="shared.GemChecker.NotGemable">Your {slotName} cannot take a gem.</Trans>;
+    }
+    //#endregion
+
+    //X gems Missing
+    if (ItemHelper.hasBonusId(item, ItemHelper.TRIPLE_GEM_BONUS_ID)) {
+      const missingGems = 3 - (item.gems?.length ?? 0);
+      if (missingGems > 0) {
+        tooltipContent.push(
+          <Trans id="shared.GemChecker.MissingGemsMultiple">
+            You are missing {missingGems} gems on your {slotName}.
+          </Trans>,
+        );
+      } else {
+        tooltipContent.push(
+          <Trans id="shared.GemChecker.guide.FullyGemmed">
+            {slotName} slots are fully gemmed!
+          </Trans>,
+        );
+      }
+    } else if (ItemHelper.hasBonusId(item, ItemHelper.DOUBLE_GEM_BONUS_ID)) {
+      const missingGems = 2 - (item.gems?.length ?? 0);
+      if (missingGems > 0) {
+        tooltipContent.push(
+          <Trans id="shared.GemChecker.MissingGemsMultiple">
+            You are missing {missingGems} gems on your {slotName}.
+          </Trans>,
+        );
+      } else {
+        tooltipContent.push(
+          <Trans id="shared.GemChecker.guide.FullyGemmed">
+            {slotName} slots are fully gemmed!
+          </Trans>,
+        );
+      }
+    } else if (ItemHelper.hasBonusId(item, ItemHelper.SINGLE_GEM_BONUS_ID)) {
+      if ((item.gems?.length ?? 0) !== 1) {
+        tooltipContent.push(
+          <Trans id="shared.GemChecker.MissingGemsSingle">
+            You are missing a gem on your {slotName}.
+          </Trans>,
+        );
+      } else {
+        tooltipContent.push(
+          <Trans id="shared.GemChecker.guide.FullyGemmed">
+            {slotName} slots are fully gemmed!
+          </Trans>,
+        );
+      }
+    }
+
+    //X Missing Setting
+    if (GemChecker.twoAddableGemSlots.includes(slotNumber)) {
+      const missingGems = 2 - (item.gems?.length ?? 0);
+      if (missingGems > 0) {
+        tooltipContent.push(
+          <Trans id="shared.GemChecker.MissingSlotsCraftable">
+            You are missing {missingGems} possible slot on your {slotName}.<br />
+            Craft/Buy Magnificent Jeweler's Setting to add a slot per.
+          </Trans>,
```

Comment:
don't use `<br/>` for formatting. it doesn't play nicely with variable screen widths.

```suggestion
          <Trans id="shared.GemChecker.MissingSlotsCraftable">
            <p>You are missing {missingGems} possible slot on your {slotName}.</p>
            <p>Craft/Buy <ItemLink id={ITEMS.MAGNIFICENT_JEWELERS_SETTING.id} /> to add slots.</p>
          </Trans>,
```

Reply:
What would you recommend for a newline?

Reply:
I see it.  You would suggest a paragraph wrap ok.  That use to be frowned on for this case.